input.c

```diff
@@ -431,10 +462,10 @@ static void ip_reset(struct input_plugin *ip, int close_fd)
 	ip_init(ip, ip->data.filename);
 	if (fd != -1) {
 		if (close_fd)
-			close(fd);
+			connection_close(get_connection(ip));
 		else {
 			lseek(fd, 0, SEEK_SET);
```

Comment:
How would this work? Your trying to do a lseek on a socket and/or ssl socket stream?